subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/SwiftXCTestErrorHandlingIntegrationTest.groovy

```diff
@@ -0,0 +1,130 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.test.xctest
+
+import org.gradle.integtests.fixtures.DefaultTestExecutionResult
+import org.gradle.integtests.fixtures.TestExecutionResult
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
+import org.gradle.nativeplatform.fixtures.RequiresInstalledToolChain
+import org.gradle.nativeplatform.fixtures.ToolChainRequirement
+import org.gradle.nativeplatform.fixtures.app.SwiftAppWithLibrariesAndXCTest
+import org.gradle.nativeplatform.fixtures.app.XCTestCaseElement
+import org.gradle.nativeplatform.fixtures.app.XCTestSourceElement
+import org.gradle.nativeplatform.fixtures.app.XCTestSourceFileElement
+
+import static org.gradle.util.Matchers.containsText
+
+@RequiresInstalledToolChain(ToolChainRequirement.SWIFTC)
+class SwiftXCTestErrorHandlingIntegrationTest extends AbstractInstalledToolChainIntegrationSpec {
+
+    def "fails when application cannot load shared library at runtime"() {
+        buildWithApplicationAndDependencies()
+        buildFile << """
+            project(':app') {
+                tasks.withType(XCTest).configureEach {
+                    doFirst {
+                        delete project(':hello').layout.buildDir.get()
+                    }
+                }
+            }
+        """
+
+        expect:
+        fails(':app:test')
+
+        and:
+        failure.assertHasCause("There were failing tests.")
+        def testFailure = testExecutionResult.testClass("Gradle Test Run :app:xcTest")
+        testFailure.assertTestFailed("execution failure", containsText("Failure while running xctest executable"))
+        if (OperatingSystem.current().isMacOsX()) {
+            testFailure.assertStderr(containsText("The bundle “AppTest.xctest” couldn’t be loaded because it is damaged or missing necessary resources"))
+        } else {
+            testFailure.assertStderr(containsText("cannot open shared object file"))
+        }
+
+    }
+
+    def "fails when force-unwrapping an optional results in an error"() {
+        buildWithApplicationAndDependencies()
+        addForceUnwrappedOptionalTest()
+
+        expect:
+        fails(':app:test')
+
+        and:
+        failure.assertHasCause("There were failing tests.")
+        testExecutionResult.testClass("ForceUnwrapTestSuite").assertTestFailed("testForceUnwrapOptional", containsText("Failure while running xctest executable"))
```

Comment:
It looks like we now add two artificial tests with this failure:

![image](https://user-images.githubusercontent.com/5387972/40499428-e83e7124-5f4f-11e8-9269-1ca1073799d3.png)

If we've started to execute tests, I think we shouldn't show the root failure. I think this would mean every build with failed tests would have the "Gradle Test Run" suite as a failure?

We also shouldn't attach the "execution failed"  to the test suite.  This makes it look like we have more tests than we actually do and tests that don't actually exist.

I think it's good that the tests in progress have some kind of failure attached to them, but it's not really useful. Running the tests by hand, I don't see any better output, so I think we're at the mercy of XCTest. I couldn't find any way to get more information about why the process stopped/failed and the one error message (Illegal Instruction 4) doesn't seem to be a part of the process's stdout or stderr.

Reply:
I was thinking that it would make it clearer that test execution stopped before all tests could be run, but I think we're constrained by what we can express in the report.  I'll change it so that it just associates the failure with the last test executed.

Reply:
This has been changed so that only the currently running test gets the failure (or the root test suite if no tests have run).